pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -354,11 +355,36 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// Remove dead container log symlinks.
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
+	var containers []*runtimeapi.Container
+	if len(logSymlinks) > 0 {
+		var err error
+		if containers, err = cgc.manager.getKubeletContainers(true); err != nil {
+			klog.V(5).Infof("unable to obtain containers: %v", err)
+			return nil
+		}
+	}
+outer:
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+			if containerID, err := getContainerIDFromLegacyLogSymlink(logSymlink); err == nil {
+				for _, container := range containers {
+					if strings.HasPrefix(container.Id, containerID) {
+						if container.State != runtimeapi.ContainerState_CONTAINER_EXITED {
+							// the container still exists, don't remove the symlink
+							klog.V(5).Infof("container still exists, not removing symlink: %s", logSymlink)
+							continue outer
+						}
+						break
+					}
+				}
+			} else {
+				klog.V(5).Infof("unable to obtain container Id: %v", err)
```

Comment:
Maybe log with higher level? This doesn't seem to be a normal case.

Reply:
Changed to level 4